aiidalab_widgets_base/nodes.py

```diff
@@ -210,22 +215,30 @@ def _find_called(cls, root):
     @classmethod
     def _find_outputs(cls, root):
         process_node = load_node(root.parent_pk)
-        if root.namespace:
-            outputs = process_node.outputs[root.namespace]
-        else:
-            outputs = process_node.outputs
 
+        outputs = process_node.outputs
+        # if output tree node is from namespace node
+        namespaces = root.namespaces
+        if namespaces:
+            outputs = functools.reduce(
+                lambda d, namespace: d[namespace], namespaces, process_node.outputs
+            )
+
+        # convert aiida LinkManager to type dict
         output_nodes = {key: outputs[key] for key in outputs}
 
         for key in sorted(
             output_nodes.keys(), key=lambda k: getattr(outputs[k], "pk", -1)
         ):
             node = output_nodes[key]
-
             if isinstance(node, AttributeDict):
+                # for namespace tree node attach label and continue recursively
                 yield AiidaOutputsTreeNode(
-                    name=key, parent_pk=root.parent_pk, namespace=key
+                    name=key,
+                    parent_pk=root.parent_pk,
+                    namespaces=(*namespaces, key),
```

Comment:
In combination with above suggestion:
```suggestion
                    namespaces=(*root.namespaces, key),
```